pilot/pkg/security/authz/policy/v1alpha1/v1alpha1_test.go

```diff
@@ -198,3 +199,68 @@ func TestV1alpha1Generator_Generate(t *testing.T) {
 		})
 	}
 }
+
+func TestV1alpha1_TrustDomainAliases(t *testing.T) {
```

Comment:
It's confusing looking at this as it's different from the naming convention used in the unit test, This is testing the `Generate()` function so could you merge this to the existing `TestV1alpha1Generator_Generate()`?

Reply:
The current way TestV1alpha1Generator_Generate is tested is different than what's needed to test rbac config with trust domain aliases. I feel like this will require some work to rewrite the checking rule that TestV1alpha1Generator_Generate is using, and since the code cut is very near, I'm not sure if it's possible. Maybe add a TODO to merge them later?

Reply:
You don't need to make too much code change here, you probably just need to put your `expectPrincipals` into the `wantRules` field. 

Take a closer look at the TestV1alpha1Generator_Generate checking rule it's essentially doing the same thing  of this TestV1alpha1_TrustDomainAliases, it's also testing more more things and gives you better coverage, for example the role name and counts in the rules.

Reply:
You're right. Thanks!